Userland/Libraries/LibWebView/CookieJar.cpp

```diff
@@ -95,15 +96,23 @@ String CookieJar::get_cookie(const URL::URL& url, Web::Cookie::Source source)
         return {};
 
     auto cookie_list = get_matching_cookies(url, domain.value(), source);
+
+    // 4. Serialize the cookie-list into a cookie-string by processing each cookie in the cookie-list in order:
     StringBuilder builder;
 
     for (auto const& cookie : cookie_list) {
-        // If there is an unprocessed cookie in the cookie-list, output the characters %x3B and %x20 ("; ")
         if (!builder.is_empty())
             builder.append("; "sv);
 
-        // Output the cookie's name, the %x3D ("=") character, and the cookie's value.
-        builder.appendff("{}={}", cookie.name, cookie.value);
+        // 1. If the cookies' name is not empty, output the cookie's name followed by the %x3D ("=") character.
+        if (!cookie.name.is_empty())
+            builder.appendff("{}=", cookie.name);
+
+        // 2. If the cookies' value is not empty, output the cookie's value.
+        if (!cookie.value.is_empty())
+            builder.append(cookie.value);
+
+        // 3. If there is an unprocessed cookie in the cookie-list, output the characters %x3B and %x20 ("; ").
```

Comment:
Should the `if (!builder.is_empty())` condition at the start of the loop be moved here?

Reply:
No - unprocessed means there's another cookie to append, i.e. we're not at the end of the list yet. So I left the condition at the top, instead of switching to an index-based loop and checking `if (i < size - 1)` here.